src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -8,51 +8,86 @@
-    CDSWheelControllerTwo = new MotorController("Follows CDS Wheel Controller", Constants.kCDSMotorFiveID);
-    
-    CDSWheelControllerTwo.getSparkMax().follow(CDSWheelControllerOne.getSparkMax());
+    CDSBeltController = new MotorController("CDS Motor", Constants.CDSBeltID);
+    CDSMotorController2 = new MotorController("CDS Motor 2", Constants.CDSMotorID);
+   
+    intitalBallSensor = new DigitalInput(Constants.ballSensorChannel);
+    middleBallSensor = new DigitalInput(Constants.ballSensorChannel);
+    finalBallSensor = new DigitalInput(Constants.ballSensorChannel);
   }
 
-  public void HopperSwitch(boolean on) {
+  public void CDSSwitch(boolean on) {
     if (on) {
-      double beltSmartSpeed = SmartDashboard.getNumber("Belt Speed", Constants.kCDSBeltSpeed);
-      double wheelSmartSpeed = SmartDashboard.getNumber("Wheel Speed", Constants.kCDSWheelSpeed);
-
-      CDSBeltController.getSparkMax().set(beltSmartSpeed);
-      CDSWheelControllerOne.getSparkMax().set(wheelSmartSpeed);
-      SmartDashboard.putNumber("CDS Belt Speed", Constants.kCDSBeltSpeed);
-      SmartDashboard.putNumber("CDS Wheel Speed", Constants.kCDSWheelSpeed);
+      CDSBeltController.getSparkMax().set(Constants.CDSBeltSpeed);
+      SmartDashboard.putNumber("CDS Belt Speed", Constants.CDSBeltSpeed);
     } else {
       CDSBeltController.getSparkMax().set(0.0);
-      CDSWheelControllerOne.getSparkMax().set(0.0);
       SmartDashboard.putNumber("CDS Motor Speed", 0);
     }
   }
 
   public void ForwardCDS() {
-    CDSBeltController.getSparkMax().setInverted(false);
-    CDSWheelControllerOne.getSparkMax().setInverted(false);
+    CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
     SmartDashboard.putString("CDS Belt Direction", "Forward");
-    SmartDashboard.putString("CDS Wheel Direction", "Forward");
   }
 
   public void ReverseCDS() {
-    CDSBeltController.getSparkMax().setInverted(true);
-    CDSWheelControllerOne.getSparkMax().setInverted(true);
+    CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
     SmartDashboard.putString("CDS Belt Direction", "Reverse");
-    SmartDashboard.putString("CDS Wheel Direction", "Reverse");
   }
 
+  public boolean getDirection() {
+    // true = inverted, false = forward
+    if (CDSBeltController.getSparkMax().get() > 0) {
+      return false;
+    } else {
+    return CDSBeltController.getSparkMax().getInverted();
+    }
+  }
+
+  public boolean getInitialSensorStatus(){
+    return intitalBallSensor.get();
+  }
+  public boolean getMiddleSensorStatus(){
+    return middleBallSensor.get();
+  }
+  public boolean getFinalSensorStatus(){
+    return finalBallSensor.get();
+  }
+  public void indexBall() {
+    boolean initialSensorStatus = getInitialSensorStatus();
+    boolean finalSensorStatus = getFinalSensorStatus();
+    if (!initialSensorStatus)  {
+      if (!this.getDirection()){
+        ballCount++;
+      } else {
+        ballCount--;
+      }
+      if (ballCount > 2) {
```

Comment:
I would lean towards moving the logic to take action on having too many cargo into a different method, instead of making it part of indexBall().